lib/buffer.js

```diff
@@ -44,27 +45,77 @@ function alignPool() {
 }
 
 
-function Buffer(arg, encoding) {
+function Buffer(arg, encodingOrOffset, length) {
   // Common case.
   if (typeof arg === 'number') {
-    // If less than zero, or NaN.
-    if (arg < 0 || arg !== arg)
-      arg = 0;
-    return allocate(arg);
-  }
-
-  // Slightly less common case.
-  if (typeof arg === 'string') {
-    return fromString(arg, encoding);
+    if (typeof encodingOrOffset === 'string') {
+      throw new Error(
+        'If encoding is specified then the first argument must be a string'
```

Comment:
Ok, pull that throw back out and another. It's still not 100% "purely additive" in that it includes the change to `fill('')` to ensure it zero-fills as opposed to doing nothing.